rasa/shared/core/events.py

```diff
@@ -301,8 +305,24 @@ def resolve_by_type(
             raise ValueError(f"Unknown event name '{type_name}'.")
 
     def apply_to(self, tracker: "DialogueStateTracker") -> None:
+        """Applies event to current conversation state.
+
+        Args:
+            tracker: The current conversation state.
+        """
         pass
 
+    def __eq__(self, other: Any) -> bool:
+        """Compares object with other object."""
+        if not isinstance(other, self.__class__):
+            return NotImplemented
+
+        return True
```

Comment:
this doesn't seem right, shouldn't this check some attributes? any two events will be equal in this case

Reply:
I see that it is overwritten in most places and correct for the places where it isn't but it does seem like a dangerous default

Reply:
@wochinge could you please take this over

Reply:
only events of the same type will be equal. It was like that before - I just removed the duplication. We can raise a `NotImplementedError` and hence force events to override it properly but that's in my opinion a change which should be done on `master` as it's not related to this PR.

Reply:
I mean there was no previous implementation on the base class, if it isn't testing for attributes it should fall back to a save default like checking for object identity (`self is other`)

Reply:
Ok